halo2/src/circuit_builder.rs

```diff
@@ -181,22 +184,25 @@ impl<'a, T: FieldElement, F: PrimeField<Repr = [u8; 32]>> Circuit<F> for PowdrCi
         meta.enable_equality(config.instance);
 
         // Add polynomial identities
-        meta.create_gate("main", |meta| -> Vec<(String, Expression<F>)> {
-            analyzed
-                .identities_with_inlined_intermediate_polynomials()
-                .iter()
-                .filter_map(|id| match id.kind {
-                    IdentityKind::Polynomial => {
+        let identities = analyzed
+            .identities_with_inlined_intermediate_polynomials()
+            .into_iter()
+            .filter(|id| id.kind == IdentityKind::Polynomial)
+            .collect::<Vec<_>>();
+        if !identities.is_empty() {
```

Comment:
Before, constructing the Halo2 circuit would fail if there is no polynomial identity.